File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.java
class Solution {
    public int largestInteger(int[] nums, int k) {
        int ans=-1;
        int max=nums[0];
        int n=nums.length;
        Map<Integer,Integer> mp=new HashMap<>();
        for(int v:nums){
            max=Math.max(max,v);
            mp.put(v,mp.getOrDefault(v,0)+1);
        }
        if(k==n)
        {ans= max;
        }
        else if(k==1){
            for(Map.Entry<Integer,Integer> entry:mp.entrySet()){
                if(entry.getValue()==1)ans=Math.max(ans,entry.getKey());
            }
        }
        else if(k>1 && k<n){
            int t1=mp.get(nums[0]);
            int t2=mp.get(nums[n-1]);
            if(t1==1 && t2==1){
                ans=Math.max(nums[0],nums[n-1]);
            }else if(t1==1){
                ans=nums[0];
            }else if(t2==1){
                ans=nums[n-1];
            }
        }
        return ans;
    }
}

File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        fq={}
        for i in nums:
            fq[i]=fq.get(i,0)+1
        ans=-1
        if k==len(nums):
            ans= max(nums)
        elif k==1:
            for k,v in fq.items():
                if(v==1):
                    ans=max(ans,k)
            
        elif k>1 and k <len(nums):
            t1=fq.get(nums[0])
            t2=fq.get(nums[-1])
            if(t1==1 and t2==1):
                ans= max(nums[0],nums[-1])
            elif t1==1:
                ans= nums[0]
            elif t2==1:
                ans= nums[-1]
        return ans